Sources/Commands/SwiftTool.swift

```diff
@@ -475,24 +488,67 @@ public class SwiftTool {
 
     }
 
-    private func editsDirectory() throws -> AbsolutePath {
+    /// Returns the currently active workspace.
+    func getActiveWorkspace() throws -> Workspace {
+        if let workspace = _workspace {
+            return workspace
+        }
+
+        // we do not take an observabilityScope in the workspace initializer as the workspace is designed to be long lived
+        // instead observabilityScope is passed into the individual workspace methods which are short lived
+        // however, the workspace initializer may need to emit diagnostics, for which the CLI sets this thread local
+        Thread.current.threadDictionary["observabilityScope"] = observabilityScope
```

Comment:
@abertelrud wdyt about this? see comments for context

Reply:
Conceptually this seems like a good direction.

Code-wise it's probably better to have an extension on `Thread` to set the top-level scope rather than hardcoding the string, but that's a detail.  I expect that over time what we'll want is a top-level scope on the action, not the object, but this is a good start.